api/provider/aws/task_delete.go

```diff
@@ -13,12 +14,12 @@ import (
 func (t *TaskProvider) Delete(taskID string) error {
 	environmentID, err := lookupEntityEnvironmentID(t.TagStore, "task", taskID)
 	if err != nil {
-		return err
+		log.Printf("[WARN] Environment not found\n")
```

Comment:
We need to do type assertions on the error and ensure that the error code is `TaskDoesNotExist`. Only in that scenario do we `return nil` early (also, you forgot to `return nil` after logging) .

Reply:
Okay and this behavior is already done here right https://github.com/quintilesims/layer0/blob/e6cf881ab637f2f5cd57848acc639fa262251b74/api/provider/aws/task_delete.go#L48

Reply:
similar behavior, but you should be checking for an `*errors.ServerError` and checking the `.StatusCode`, not the message